controllers/infrastructure/byomachine_controller.go

```diff
@@ -302,6 +323,12 @@ func (r *ByoMachineReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 			&source.Kind{Type: &infrav1.ByoHost{}},
 			handler.EnqueueRequestsFromMapFunc(ByoHostToByoMachineMapFunc(controlledTypeGVK)),
 		).
+		Watches(
+			&source.Kind{Type: &infrav1.K8sInstallerConfig{}},
```

Comment:
We can't watch to K8sInstallerConfig as installer is pluggable. 
I guess we need to do something like clusterapi: https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/controllers/machine/machine_controller_phases.go#L150

Reply:
IMO Watch is not required for the installer config object. We can fetch and force a requeue after some predefined seconds if the status is not ready. No other events before or after required reconcile from byomachine controller.